### PR TITLE
Exercise 5

### PR DESCRIPTION
**Pull request by slipstreamamx**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+FROM base as development
+
+COPY docker-entrypoint-dev.sh /app
+
+ENTRYPOINT ["./docker-entrypoint-dev.sh"]
+
+FROM base as production
+
+COPY docker-entrypoint-prod.sh /app
+RUN chmod +x /var/log
```
Comment (BenRamchandani) [inline Dockerfile]: What's this for?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+ENV PYTHONPATH=${PYTHONPATH}:${PWD} 
+
+WORKDIR /app
+
+COPY poetry.lock pyproject.toml /app
+
+RUN pip3 install poetry
+
+EXPOSE 5001
```
Comment (BenRamchandani) [inline Dockerfile:11]: It's just informational, so doesn't really matter, but both of your containers run on port 5000, not 5001

File: README.md [hunk 3]
```diff
@@ -121,3 +121,43 @@ $ ssh USERNAME@USERNAME@IP-ADDRESS ## this is the managed node IP address. Once
...
+The local:container ports will be mapped to 5000:5000 by default
+
+To build production images and containers, do the following:
+
+```
+$ docker-compose up production
+```
+Above builds the "production" stage from the Dockerfile, and docker-compose.yml tags the image as "todo-app:prod".
+
+The local:container ports will be mapped to 5001:5000 by default
```
Comment (BenRamchandani) [inline README.md]: This is the other way around in your Docker compose file (dev -> 5001, prod -> 5000)

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  production:
+    build:
+      context: .
+      target: production
+    image: todo-app:prod
+    container_name: todo_app_prod
+    ports:
+      - "5000:5000"
+    volumes:
+      - ./todo_app:/app/todo_app
```
Comment (BenRamchandani) [inline docker-compose.yml]: No need for the volume in the production one

**Review by BenRamchandani — APPROVED** [review]
Nice work! I've left a few optional comments on things you could clear up, but it all worked for me following the instructions in your README.

**Comment by slipstreamamx** [comment]
Thanks, Ben incorporating your suggestions.

**Comment by slipstreamamx** [comment]
thanks Ben Incorporate your change

**Pull request merged**